File: ai-backend/models/entity/user_tool_config.py
"""
用户工具配置实体类，对应表 `user_tool_config`
"""
from typing import Optional, Dict, Any
from datetime import datetime
from models.entity.base import BaseEntity


class UserToolConfig(BaseEntity):
    """用户工具配置实体类"""
    id: Optional[str] = None  # 用户工具配置id（UUID）
    user_id: Optional[str] = None  # 用户id（UUID）
    tool_name: Optional[str] = None  # 工具名称
    config_json: Optional[Dict[str, Any]] = None  # 工具配置JSON数据
    
    class Config:
        from_attributes = True

File: backend/src/main/java/top/smartduck/ducktodo/model/entity/UserToolConfig.java
package top.smartduck.ducktodo.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 用户工具配置实体类，对应表 `user_tool_config`。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("user_tool_config")
public class UserToolConfig implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 用户工具配置id（UUID）
     */
    @TableId(value = "id", type = IdType.ASSIGN_UUID)
    private String id;

    /**
     * 用户id（UUID）
     */
    private String userId;

    /**
     * 工具名称
     */
    private String toolName;

    /**
     * 工具配置JSON数据
     */
    private String configJson;

    /**
     * 是否删除 0-未删除 1-已删除
     */
    @TableLogic
    private Integer isDelete;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
